newIDE/app/src/EventsBasedObjectEditor/EventBasedObjectChildrenEditor.js

```diff
@@ -0,0 +1,278 @@
+import Window from '../Utils/Window';
+import ObjectEditorDialog from '../ObjectEditor/ObjectEditorDialog';
+import { type ObjectEditorTab } from '../ObjectEditor/ObjectEditorDialog';
+
+const gd: libGDevelop = global.gd;
+
+type Props = {|
+  project: gdProject,
+  globalObjectsContainer: gdObjectsContainer,
+  eventsFunctionsExtension: gdEventsFunctionsExtension,
+  eventsBasedObject: gdEventsBasedObject,
+|};
+
+type State = {|
+  editedObjectWithContext: ?ObjectWithContext,
+  editedObjectInitialTab: ?ObjectEditorTab,
+|};
+
+export default class EventBasedObjectChildrenEditor extends React.Component<
+  Props,
+  State
+> {
+  _objectsList: ?ObjectsList;
+
+  state = {
+    editedObjectWithContext: null,
+    editedObjectInitialTab: 'properties',
+  };
+
+  _onDeleteObject = (i18n: I18nType) => (
+    objectWithContext: ObjectWithContext,
+    done: boolean => void
+  ) => {
+    const { object } = objectWithContext;
+    const { project, globalObjectsContainer, eventsBasedObject } = this.props;
+
+    const answer = Window.showYesNoCancelDialog(
+      i18n._(
+        t`Do you want to remove all references to this object in groups and events (actions and conditions using the object)?`
+      )
+    );
+
+    if (answer === 'cancel') return;
+    const shouldRemoveReferences = answer === 'yes';
+
+    gd.WholeProjectRefactorer.objectOrGroupRemovedInEventsBasedObject(
+      project,
+      eventsBasedObject,
+      globalObjectsContainer,
+      eventsBasedObject,
+      object.getName(),
+      /* isObjectGroup=*/ false,
+      shouldRemoveReferences
+    );
+    done(true);
+  };
+
+  _canObjectOrGroupUseNewName = (newName: string, i18n: I18nType) => {
+    const { eventsBasedObject } = this.props;
+
+    if (
+      eventsBasedObject.hasObjectNamed(newName) ||
+      eventsBasedObject.getObjectGroups().has(newName)
+    ) {
+      showWarningBox(
+        i18n._(t`Another object or group with this name already exists.`),
+        {
+          delayToNextTick: true,
+        }
+      );
+      return false;
+    } else if (!gd.Project.validateName(newName)) {
+      showWarningBox(
+        i18n._(
+          t`This name is invalid. Only use alphanumeric characters (0-9, a-z) and underscores. Digits are not allowed as the first character.`
+        ),
+        { delayToNextTick: true }
+      );
+      return false;
+    }
+
+    return true;
+  };
+
+  _onRenameEditedObject = (newName: string, i18n: I18nType) => {
+    const { editedObjectWithContext } = this.state;
+
+    if (editedObjectWithContext) {
+      this._onRenameObject(editedObjectWithContext, newName, () => {}, i18n);
+    }
+  };
+
+  _onRenameObject = (
+    objectWithContext: ObjectWithContext,
+    newName: string,
+    done: boolean => void,
+    i18n: I18nType
+  ) => {
+    // TODO EBO Use a constant instead a hard coded value "Object".
+    if (newName === 'Object') {
+      showWarningBox(i18n._(t`"Object" is a reserved name for the parent.`), {
```

Comment:
is a reserved name, used for the parent object in the events (actions, conditions, expressions...). Please choose another name.

(for error message, state the problem and then what to do/how to fix the problem)